### ISSUE TITLE
Fix S2259 FN: Rule is not executed

### ISSUE THREAD

**Issue by zsolt-kolbay-sonarsource** [SonarSource/sonar-dotnet#6904]
### Description

The `ShouldExecute()` method of S2259 (Roslyn SE implementation) is returning `false` in certain cases when it shouldn't: the `Result` variable is set to `false` when a conditional access or a coalesce expression is encountered in the syntax tree. This automatically aborts any further exploration of the syntax tree and the `ShouldExecute()` method will immediately return `false`.
That means that any method which contains a conditional access operation will not be validated, even if there's potential NRE right after that:

### Repro steps

```
class Test
{
    void ConditionalAccess()
    {
        string str = null;
        _ = str?.Length;
        int length = str.Length; // FN - this will throw a NRE
    }
}
```

### Expected behavior

An issue should be raised on the last line of the method.

### Actual behavior

No issue is raised because the rule is not executed.

### Known workarounds

None.

### Related information

* C#/VB.NET Plugins version: 8.54

**Comment by pavel-mikula-sonarsource** [comment]
Duplicate #4250
